asf_public_discourse_home_decarbonisation/analysis/buildhub/extract_faqs.py

```diff
@@ -0,0 +1,245 @@
+"""
+This script is designed to extract frequently asked questions (FAQs) and specific patterns from a given forum text dataset.
+Key Functionalities:
+    - Extract Questions: Identifies and extracts questions from text based on criteria such as the number of words and minimum sentence length.
+    - Extract Specific Patterns: Targets and extracts specific patterns or phrases, tailored by the `extract_idk` function.
+    - Configurable: Allows runtime configuration through command-line arguments for flexible usage in different scenarios.
+To run this script:
+    'python extract_faqs.py --forum <forum> --category <category> --collection_date_time <collection_date_time>'
+where:
+    <forum>: The forum that we are looking at (e.g. bh or MSE).
+    <category>: The category within the forum (e.g. energy).
+    <collection_date_time>: The date time format, this differs between 'YYYY_MM_DD' for MSE and 'YYMMDD' for bh.
+Example Usage:
+    For MSE:
+        'python extract_faqs.py --forum mse --category energy --collection_date_time 2023_11_15'
+    For buildhub:
+        'python extract_faqs.py --forum bh --category combined_data --collection_date_time 231120'
+    Note: In these run commands note the different format in date-time format for the different forums.
+"""
+import pandas as pd
+import re
+import nltk
+from nltk.tokenize import sent_tokenize
+from typing import List, Tuple
+import argparse
+from asf_public_discourse_home_decarbonisation.getters.bh_getters import (
+    get_bh_category_data,
+)
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import (
+    get_mse_category_data,
+)
+import os
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "119_air_source_heat_pumps_ashp")
+    - collection_date_time: collection date/time (defaults to "231120")
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--forum",
+        help="Forum that we are looking at",
+        default="bh",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="119_air_source_heat_pumps_ashp",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--collection_date_time",
+        help="Collection date/time",
+        default="231120",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def extract_questions(
+    sentences: List[str], num_of_words: int = 5, min_length: int = 5
```

Comment:
i'd say `num_of_words` is probably enough at this point. `min_length` seems a bit random to set for now?

Reply:
As mentioned above, the number of words condition only comes into play if the question word is in the sentence, otherwise we could be dealing with questions like "no?", "yes?", "maybe?". I'm going to think about this a bit more in terms of the methodology.